internal/handler/middleware/middleware.go

```diff
@@ -33,6 +34,7 @@ type MsgRequest struct {
 // MsgResponse represent outgoing response to the client.
 type MsgResponse struct {
 	*wire.OpMsg
+	*mongoerrors.Error
```

Comment:
```suggestion
	err *mongoerrors.Error
```

Embedding two fields could lead to clashes of method sets. And we don't actually want to embed any of them; we just embedded OpMsg to reduce the number of changes in handlers for now.

Reply:
That's true. Newly added fields doesn't need to be embedded.